publish-release.sh

```diff
@@ -30,40 +31,52 @@ EOD
 }
 
 # get the options
-while getopts h c ; do
+while getopts hm c ; do
     case $c in
 	    h) usage ; exit 0 ;;
+            m) minimal=true ;;
 	    \?) usage ; exit 2 ;;
     esac
 done
 shift `expr $OPTIND - 1`
 
-if [ $# -ne 1 ] ; then
+if [ $# -gt 1 ] ; then
     usage
     exit 2
 fi
 
-releasetag="$1"
-export OP_VERSION="$releasetag"
+
+if [ $# -eq 1 ] ; then
+  releasetag="$1"
+  export OP_VERSION="$1"
+  message="Publish release"
+else
+  message="Publish version"
+fi
 
 . "$DIR/env.build.sh"
 
-make yaml yaml-ns-scoped
 $DIR/build.sh
+$DIR/push-image.sh
+make yaml yaml-ns-scoped
 
-echo "Update Qserv images in manifests/base/image.yaml"
-sed -ri  "s/^(\s*image: qserv\/.*:).*/\1$releasetag/" $DIR/manifests/base/image.yaml
-echo "Update release number in documentation"
-find $DIR/doc -type f -print0 | xargs -0 sed -ri  "s/RELEASE=\".*\"/RELEASE=\"$releasetag\"/"
-sed -ri  "s/RELEASE=\".*\"/RELEASE=\"$releasetag\"/" $DIR/README.md
+if [ $minimal = false ]; then
 
-# Prepare operatorHub files
-# Edit 'replaces', 'image' and 'containerImage' fields in config/manifests/bases/qserv-operator.clusterserviceversion.yaml
-csv_file="$DIR/config/manifests/bases/qserv-operator.clusterserviceversion.yaml"
-sample_file="$DIR/config/samples/qserv_v1beta1_qserv.yaml"
-previous_version=$(grep -oP 'qserv\/qserv-operator:([0-9]+\.[0-9]+\.[0-9](-rc[0-9]+)?)' "$csv_file" | cut -d: -f2)
-sed -ri  "s/(202[0-9]+\.[0-9]+\.[0-9](-rc[0-9]+)?)/$releasetag/" "$sample_file" "$csv_file"
-sed -ri  "s/replaces: qserv-operator\.v([0-9]+\.[0-9]+\.[0-9](-rc[0-9]+)?)/replaces: qserv-operator\.v$previous_version/"  "$csv_file"
+  echo "Update Qserv images in manifests/base/image.yaml"
+  sed -ri  "s/^(\s*image: qserv\/.*:).*/\1$releasetag/" $DIR/manifests/base/image.yaml
+  echo "Update release number in documentation"
+  find $DIR/doc -type f -print0 | xargs -0 sed -ri  "s/RELEASE=\".*\"/RELEASE=\"$releasetag\"/"
+  sed -ri  "s/RELEASE=\".*\"/RELEASE=\"$releasetag\"/" $DIR/README.md
+
+  # Prepare operatorHub files
+  # Edit 'replaces', 'image' and 'containerImage' fields in config/manifests/bases/qserv-operator.clusterserviceversion.yaml
+  csv_file="$DIR/config/manifests/bases/qserv-operator.clusterserviceversion.yaml"
+  sample_file="$DIR/config/samples/qserv_v1beta1_qserv.yaml"
+  previous_version=$(grep -oP 'qserv\/qserv-operator:([0-9]+\.[0-9]+\.[0-9](-rc[0-9]+)?)' "$csv_file" | cut -d: -f2)
+  sed -ri  "s/(202[0-9]+\.[0-9]+\.[0-9](-rc[0-9]+)?)/$releasetag/" "$sample_file" "$csv_file"
```

Comment:
So, this is not expected to last until 2030?

Reply:
No, I hope it will be much more simpler in 2030 ;-)